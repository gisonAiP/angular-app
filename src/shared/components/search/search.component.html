<div class="d-none d-md-flex py-5 align-items-center">
  <img src="../../../assets/svg/search.svg" alt="search" height="20" class="px-1"/>
  <input class="input-search autocomplete m-2" placeholder="Search..."/>
  <input class="input-search country mx-2" placeholder="Poland"/>
  <button data-test="search-button" class="btn-search mx-2">Search</button>
</div>
