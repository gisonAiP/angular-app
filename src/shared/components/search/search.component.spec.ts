import { SearchComponent } from './search.component';
import { createRoutingFactory } from '@ngneat/spectator';
import { Spectator } from '@ngneat/spectator/jest';


describe('SearchComponent', () => {
  let spectator: Spectator<SearchComponent>;
  const createComponent = createRoutingFactory({
    component: SearchComponent,
    shallow: true,
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should validate snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should display button', () => {
    expect(spectator.query('[data-test="search-button"]').textContent).toEqual('Search');
  });
});

