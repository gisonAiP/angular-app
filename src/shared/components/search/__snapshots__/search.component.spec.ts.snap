// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent should validate snapshot 1`] = `
<app-search>
  <div
    class="d-none d-md-flex py-5 align-items-center"
  >
    <img
      alt="search"
      class="px-1"
      height="20"
      src="../../../assets/svg/search.svg"
    />
    <input
      class="input-search autocomplete m-2"
      placeholder="Search..."
    />
    <input
      class="input-search country mx-2"
      placeholder="Poland"
    />
    <button
      class="btn-search mx-2"
      data-test="search-button"
    >
      Search
    </button>
  </div>
</app-search>
`;
